### PR TITLE
Validate config file

### LINKED ISSUE #13: Add validation for config file
We should validate configuration file and provide helpful error message if the config doesn't match provided schema.

### PR DESCRIPTION
**Pull request by mrkldshv**
Closes #13.

### CODE REVIEW

**Comment by mrkldshv** [comment]
I'm still figuring out the way to keep zod schema and TS types in sync. There're a few utilities that generate types from zod schema, however this will prevent us from keep config properties commented.
I'm thinking about constraining zod schema by types (so schema depends on types, instead of types depend on schema), but I'm still getting some type errors related to `extend` method.

**Comment by ds300** [comment]
> constraining zod schema by types

that's a fine tradeoff for me. as long as you can get it working! If you need to refactor the types themselves, go ahead.

**Comment by mrkldshv** [comment]
Sure! I also decided to have setting the defaults in the separate function because there's quite a bit of custom logic that is not possible to implement with zod's `default` schema method.

File: package.json [hunk 1]
```diff
@@ -80,8 +80,10 @@
     "prettier": "^2.8.7",
     "prettier-plugin-organize-imports": "^3.2.2",
     "rimraf": "^5.0.0",
-    "semver": "^7.4.0",
-    "typescript": "^5.0.3"
+    "typescript": "^5.0.3",
+    "zod": "^3.21.4",
+    "zod-validation-error": "^1.3.0",
```
Comment (ds300) [inline package.json:85]: these should both be in `dependencies`

**Review by ds300 — APPROVED** [review]
Wow @mrkldshv this works beautifully! Stellar work! 🥳 

Once you move the two zod packages into the main `"dependencies"` object this will be good to go!

File: package.json (hunk 1 above)
Comment (mrkldshv) [reply package.json:85]: Yeah, I wasn't sure about that. What logic do we choose to follow when deciding between dev deps/deps?

Comment (ds300) [reply package.json]: Our dev deps won't be installed when someone adds this package to their own npm project, or installs it globally.

Comment (mrkldshv) [reply package.json]: Got it. Thanks!

**Pull request merged**